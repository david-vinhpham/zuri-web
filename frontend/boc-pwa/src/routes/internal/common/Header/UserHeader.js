import React, { PureComponent } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { bool, func, any, objectOf, noop } from 'prop-types';
import { Button } from 'homecredit-ui';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { logout, checkAuth } from 'actions/internal';
import history from '../../../../history';
import s from './UserHeader.css';
import { USER_TEXT } from '../constant';

class UserHeader extends PureComponent {
  static propTypes = {
    checkAuth: func,
    logout: func,
    isCheckAuth: bool,
    auth: objectOf(any),
  };

  static defaultProps = {
    checkAuth: noop,
    logout: noop,
    isCheckAuth: false,
    auth: {},
  };

  componentWillMount() {
    const { isCheckAuth } = this.props;
    this.props.checkAuth(isCheckAuth, history);
  }

  onLogoutClick = () => {
    this.props.logout(history);
  };

  render() {
    const { auth } = this.props;
    return (
      <div>
        {auth &&
          auth.user.name && (
            <div>
              <span />
              <span className={s.welcomeText}>
                <span />
                {USER_TEXT.welcome} {auth.user.name} !
              </span>
              <Button onClick={this.onLogoutClick} className={s.btnLogout}>
                <span className={s.btnLogoutText}>{USER_TEXT.logout}</span>
              </Button>
            </div>
          )}
      </div>
    );
  }
}

function mapStateToProps({ internal }) {
  const { auth } = internal;
  return { auth };
}

const enhance = compose(
  connect(
    mapStateToProps,
    { checkAuth, logout },
  ),
  withStyles(s),
);

export default enhance(UserHeader);
