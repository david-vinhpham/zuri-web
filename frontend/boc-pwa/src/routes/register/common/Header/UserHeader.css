
@import '../../../../components/variables.css';

.welcomeText {
  width: 100%;
  margin: 0;
  text-align: center;
  background-color: var(--c-accent-color);
  font-weight: var(--title-font-weight);
  font-size: var(--f-label);
  padding: var(--b-space-md) 0;
  color: var(--c-ui-lighter);
  font-style: italic;
  padding-right: var(--b-space-xs);
}

.btnLogoutText {
  padding-right: var(--b-space-2xs);
}

.btnLogout {
  box-shadow: none;
  padding-right: var(--b-space-xs);
}

.btnLogout:active {
  background-color: transparent;
}

@media (--xs) {
  .welcome {
    display: none;
  }
}
