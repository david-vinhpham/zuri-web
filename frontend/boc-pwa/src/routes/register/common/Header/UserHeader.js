import React, { PureComponent } from 'react';
import { func, noop, string } from 'prop-types';
import { Button } from 'homecredit-ui';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import s from './UserHeader.css';
import { USER_TEXT } from '../constant';

class UserHeader extends PureComponent {
  static propTypes = {
    username: string,
    logout: func,
  };

  static defaultProps = {
    username: '',
    logout: noop,
  };

  onLogoutClick = () => {
    this.props.logout();
  };

  render() {
    const { username } = this.props;
    return (
      <React.Fragment>
        <span className={s.welcomeText}>
          <span />
          <span className={s.welcome}>{USER_TEXT.welcome}</span> {username} !
        </span>
        <Button onClick={this.onLogoutClick} className={s.btnLogout}>
          <span className={s.btnLogoutText}>{USER_TEXT.logout}</span>
        </Button>
      </React.Fragment>
    );
  }
}

export default withStyles(s)(UserHeader);
