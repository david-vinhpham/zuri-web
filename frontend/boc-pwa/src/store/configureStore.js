import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
