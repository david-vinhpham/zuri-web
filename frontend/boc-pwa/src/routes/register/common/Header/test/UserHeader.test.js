import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import UserHeader from '../UserHeader';

const mockStore = configureStore();
const initialState = {
  saRegister: {
    auth: {
      username: 'homerselect',
    },
  },
};

it('HeaderBar snapshot', () => {
  const store = mockStore(initialState);
  const wrapper = renderer.create(
    <Provider store={store}>
      <UserHeader logout={() => {}} />
    </Provider>,
  );
  const snapshot = wrapper.toJSON();

  expect(snapshot).toMatchSnapshot();
});
