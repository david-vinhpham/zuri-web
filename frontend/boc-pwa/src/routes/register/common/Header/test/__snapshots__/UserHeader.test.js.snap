// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar snapshot 1`] = `
Array [
  <span
    className="welcomeText"
  >
    <span />
    <span
      className="welcome"
    >
      Xin chào
    </span>
     
    
     !
  </span>,
  <button
    className="Button_buttonReset__3cQTl Button_button__2pQRN btnLogout"
    disabled={false}
    onClick={[Function]}
    title=""
    type="button"
  >
    <span
      className="btnLogoutText"
    >
      Đăng xuất
    </span>
  </button>,
]
`;
